src/s3/util/fileViewerManager.ts

```diff
@@ -168,20 +235,104 @@ export class S3FileViewerManager {
             }
         } else {
             //was activated from the explorer, need to get the file
+            const fileNode = uriOrNode
+
+            let headResponse
+            try {
+                headResponse = await fileNode.s3.getHeadObject({
+                    bucketName: fileNode.bucket.name,
+                    key: fileNode.file.key,
+                })
+            } catch (e) {
+                getLogger().error('S3FileViewer: Error calling getHeadObject, error: ', e)
+            }
+
+            let type
+            let charset
+            if (headResponse) {
+                type = mime.contentType(headResponse.ContentType!)
+                charset = mime.charset(type as string)
+            } else {
+                type = ''
+                charset = ''
+            }
+
             const fileLocation = await this.getFile(uriOrNode)
             if (!fileLocation) {
                 return
             }
+            const s3Uri = vscode.Uri.parse(fileLocation.fsPath)
+            let tab = this.activeTabs.get(fileLocation.fsPath)
+
+            if (!tab) {
+                tab = { fileUri: fileLocation, s3Uri, editor: undefined, s3FileNode: uriOrNode, type, charset } as S3Tab
+            }
 
-            const newTab = this.activeTabs.get(fileLocation.fsPath) ?? new S3Tab(fileLocation, uriOrNode)
-            await newTab.openFileInEditMode()
+            // Can't open images, etc. in VS Code through `openTextDocument`.
+            // HACK: Routing opening to VS Code's quickOpen functionality will use VS Code's built-in file handling
```

Comment:
Is this a hack or more of a `TODO`? Since `openFile` _should_ support more than text files, but it requires quite a bit more work to do properly.

Reply:
it is not a todo, I am just explaining the reason why on a file that is not UTF-8 I open it with the command instead of `S3FileViewer.openFile()`

Reply:
I'd say remove the line since you're no longer using `quickOpen` (thank god). You can dump the output message too.

I don't think there's any additional development to be done at this level, handing off to VS Code to deal with via `openFile` should be fine.